_inc/client/components/plans/plan-price/index.jsx

```diff
@@ -14,72 +14,90 @@ import { getCurrencyObject } from '@automattic/format-currency';
 import './style.scss';
 
 export class PlanPrice extends Component {
-	render() {
-		const { currencyCode, rawPrice, original, discounted, className } = this.props;
-
-		if ( ! currencyCode || ! rawPrice ) {
-			return null;
-		}
+	getPriceRange() {
+		const { currencyCode, rawPrice } = this.props;
 
 		// "Normalize" the input price or price range.
 		const rawPriceRange = Array.isArray( rawPrice ) ? rawPrice.slice( 0, 2 ) : [ rawPrice ];
 		if ( rawPriceRange.includes( 0 ) ) {
 			return null;
 		}
 
-		const priceRange = rawPriceRange.map( item => {
+		return rawPriceRange.map( item => {
 			return {
 				price: getCurrencyObject( item, currencyCode ),
 				raw: item,
 			};
 		} );
+	}
 
-		const classes = classNames( 'plan-price', className, {
-			'is-original': original,
-			'is-discounted': discounted,
-		} );
-
-		const renderPriceHtml = priceObj => {
-			return (
-				<>
-					<span className="plan-price__integer">{ priceObj.price.integer }</span>
-					<sup className="plan-price__fraction">
-						{ priceObj.raw - priceObj.price.integer > 0 && priceObj.price.fraction }
-					</sup>
-				</>
-			);
-		};
+	renderPrice( priceObj ) {
+		return (
+			<>
+				<span className="plan-price__integer">{ priceObj.price.integer }</span>
+				<sup className="plan-price__fraction">
+					{ priceObj.raw - priceObj.price.integer > 0 && priceObj.price.fraction }
+				</sup>
+			</>
+		);
+	}
 
-		const smallerPriceHtml = renderPriceHtml( priceRange[ 0 ] );
-		const higherPriceHtml = priceRange[ 1 ] && renderPriceHtml( priceRange[ 1 ] );
+	renderContent() {
+		const priceRange = this.getPriceRange();
+		const smallerPrice = this.renderPrice( priceRange[ 0 ] );
```

Comment:
Should we account for the cases where `getPriceRange()` returns `null` here and below?

Reply:
I'd wonder at the use case for `getPriceRange()` returning `null`. That seems to be for if the prices can include free- in that case, should we not render the component? Or rendered an i18n string equivalent of $0.00?

Reply:
Those are really good points! I thought about the case when a price value is 0 and actually wasn't sure why we don't allow it here. Since it's supported by the `getCurrencyObject` function, I don't really see a reason why we shouldn't account for 'free' products as well.

I've addressed it in cf89a32. I also removed a `currencyCode` check since we're providing a default value for this prop and updated the README file.